### DISCUSSION TITLE
Unable to Create Tag Index on Name Property

### DISCUSSION THREAD

**Discussion by bruckynetworks** [vesoft-inc/nebula#5744] [Q&A] [answered]
Hello,

I am working on a project where I am trying to create a tag index on a router tag's name property in Nebula Graph. My goal is to match network devices using their names instead of their IDs.

Here is the syntax I used:

CREATE TAG INDEX router_name_index ON router(name);

However, I encounter the following error:

[ERROR (-1005)]: Invalid param!

I have also tried specifying the data type like so:

CREATE TAG INDEX router_name_index ON router(name string);

But I received a syntax error:

[ERROR (-1004)]: SyntaxError: syntax error near `string'

I'm using Nebula Graph version 3.4 within docker-compose. Is there a correct syntax to create a tag index on a property or is what I'm attempting to do not supported? Any guidance or alternative suggestions would be greatly appreciated!

Thank you!

**Comment by wey-gu** [comment, marked as answer]
Dear @bruckynetworks !

For string prop, we need to specify the length when indexing them!

https://docs.nebula-graph.io/3.6.0/3.ngql-guide/14.native-index-statements/1.create-native-index/#create_single-property_indexes

i.e. `CREATE TAG INDEX router_name_index ON router(name string(32));`

**Reply by bruckynetworks** [reply]
Thank you so much, and apologies for the daft question.

**Reply by wey-gu** [reply]
We've all experienced those pitfalls ;) Welcome to the community!

**Reply by bruckynetworks** [reply]
Thanks for the kind works, much appreciated and even bigger thanks for the welcome to the community.
